### DISCUSSION TITLE
Find dependency to 'static library' without install

### DISCUSSION THREAD

**Discussion by MartinSchwarzbauer** [mesonbuild/meson#12791] [General]
Hi *!

I am pretty new to meson and we are gone switch our build environment to meson!
Now i have question about the following situation:

projectA - which generates a static library (this library is not installed - we just need it for compilation)
projectB - executable which uses projectA library to link

How can i setup this 'correctly'? Our IDE does the compile ... (*.a exists) but the projectB does not find this dependency, because it is not installed! How do i specify dependencies to our own local meson projects? I have no solution !

Is there any way to handle this in a proper way?

Thank you in advance,
M

**Comment by eli-schwartz** [comment]
The general way to do this is to use subprojects to build the second library as part of the same compilation:

https://mesonbuild.com/Subprojects.html#using-a-subproject

**Comment by MartinSchwarzbauer** [comment]
Thanks for your reply! But this is in our environment not possible!

We have about 20 "sub components"!  All in separate repositories! 

- projectA
- projectB
- projectC
- ...

And i wanna compile projectC, which has to link agains "projectA" and "projectB" and much more of these combinations! 
Because projectB is not part of projectA 

As you suggested, i have to maintain this in only one project with 20 subprojects ?
